### PR TITLE
Common integration scenario test

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**What this PR does / why we need it**:
This PR adds a test covering the common integration scenario.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
In order to run the test the extension must be enabled for the landscape and a shoot should exist.

**Release note**:

```other developer
An integration test covering the 'common' scenario was added.
```

### CODE REVIEW

File: test/system/lifecycle/lifecycle_test.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+			"issuerURL": oidConfig.Issuer,
+			"clientID":  clientId,
+			"keys":      base64.StdEncoding.EncodeToString(jwks),
+		}
+		err = f.RenderAndDeployTemplate(ctx, f.ShootClient, templates.CustomOIDCConfiguration, values)
+		Expect(err).ToNot(HaveOccurred())
+		getRequest := f.ShootClient.RESTClient().Get()
+		getRequest = getRequest.RequestURI("/apis/authentication.gardener.cloud/v1alpha1/openidconnects/custom")
+		getResponse := getRequest.Do(ctx)
+		Expect(getResponse.Error()).ToNot(HaveOccurred())
```
Comment (vpnachev) [inline test/system/lifecycle/lifecycle_test.go:103]: Can you use the controller runtime client `f.ShootClient.Client()` which dynamically discover resources?
It should be something like 
```golang 
oidc := openidconnectv1alpha1.OpenIDConnect{} // openidconnectv1alpha1 "github.com/gardener/oidc-webhook-authenticator/apis/authentication/v1alpha1"
err = f.ShootClient.Client().Get(ctx, types.NamespacedName{Name: "custom"}, &oidc)
Expect(err).ToNot(HaveOccurred())
```

File: test/system/lifecycle/lifecycle_test.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+			}
+		}, time.Second*20, time.Second).Should(Equal(expectedStatus))
+
+		// Delete the oidc resource
+		deleteRequest := f.ShootClient.RESTClient().Delete()
+		deleteRequest = deleteRequest.RequestURI("/apis/authentication.gardener.cloud/v1alpha1/openidconnects/custom")
+		deleteResponse := deleteRequest.Do(ctx)
+		Expect(deleteResponse.Error()).ToNot(HaveOccurred())
+		getResponse.StatusCode(&statusCode)
+		Expect(statusCode).To(Equal(200))
```
Comment (vpnachev) [inline test/system/lifecycle/lifecycle_test.go:145]: Also use the controller-runtime client can be used.

File: test/system/lifecycle/lifecycle_test.go [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+			}
+		}, time.Second*20, time.Second).Should(Equal(expectedStatus))
+
+		// Disable the OIDC extension
+		err = f.UpdateShoot(ctx, func(shoot *gardencorev1beta1.Shoot) error {
+			newExtensions := make([]gardencorev1beta1.Extension, len(shoot.Spec.Extensions)-1)
+			for _, e := range shoot.Spec.Extensions {
+				if e.Type != constants.ExtensionType {
+					newExtensions = append(newExtensions, e)
+				}
```
Comment (vpnachev) [inline test/system/lifecycle/lifecycle_test.go:169]: If the extension is globally enabled, you perhaps want to disable it with 
```golang
e.Disabled = pointer.Bool(true)
```

**Review by vpnachev — CHANGES_REQUESTED** [review]
Overall looks very good, I have some minor suggestions and questions in-line.

/invite @gardener/test-infra-maintainers

**Comment by vpnachev** [comment]
> /invite @gardener/test-infra-maintainers

Seems like the cannot be invited, so I am directly inviting the members :) 
/invite @hendrikKahl @dguendisch

File: test/system/lifecycle/lifecycle_test.go (hunk 3 above)
Comment (dimityrmirchev) [reply test/system/lifecycle/lifecycle_test.go]: I made it so that the initial extension configuration is preserved after the test execution.

**Review by dguendisch — COMMENTED** [review]
/lgtm for the testmachinery definition

File: test/system/lifecycle/lifecycle_test.go [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+				Spec: authenticationv1.TokenRequestSpec{
+					Audiences:         []string{clientId},
+					ExpirationSeconds: &ttl,
+				},
+			}, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			expectedStatus := status{
+				Status:  "Failure",
+				Message: "forbidden: User \"custom-prefix:system:serviceaccount:default:default\" cannot get path \"/\"",
```
Comment (vpnachev) [inline test/system/lifecycle/lifecycle_test.go:144]: You can avoid escaping by using back-quote
```suggestion
				Message: `forbidden: User "custom-prefix:system:serviceaccount:default:default" cannot get path "/"`,
```

File: test/system/lifecycle/lifecycle_test.go [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+			f.Logger.Warning(fmt.Sprintf("Cannot register the seed as identity provider. Error: %s. The verification of oidc provider registration was skipped. Continuing with test execution...", err.Error()))
+		}
+
+		// Ensure that the OIDC service is disabled in order to verify the deletion process
+		f.UpdateShoot(ctx, ensureOIDCServiceIsDisabled)
+
+		// Ensure that oidc authenticator deployment is deleted
+		err = f.SeedClient.Client().Get(ctx, client.ObjectKeyFromObject(oidcDeployment), oidcDeployment)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
```
Comment (vpnachev) [inline test/system/lifecycle/lifecycle_test.go:191]: gardener have a matcher for this error
```suggestion
		Expect(err).To(BeNotFoundError()) // you have to import "github.com/gardener/gardener/pkg/utils/test/matchers"
```

**Review by vpnachev — CHANGES_REQUESTED** [review]
Just some very minor change requests.

**Review by vpnachev — APPROVED** [review]
/lgtm
/squash

**Pull request merged**